From 633c3a0cae343da284394c776c67e5862d4e3b27 Mon Sep 17 00:00:00 2001
From: patch <patch@packer-output-78e360d1-5b9b-49c0-b5a7-d8eab7c3356c>
Date: Fri, 10 Apr 2026 11:50:35 +0000
Subject: [PATCH] Downgrade target

---
 meson_cpu/x86/meson.build | 15 +++++++++++----
 1 file changed, 11 insertions(+), 4 deletions(-)

diff --git a/meson_cpu/x86/meson.build b/meson_cpu/x86/meson.build
index add073376d..696ad2b648 100644
--- a/meson_cpu/x86/meson.build
+++ b/meson_cpu/x86/meson.build
@@ -3,7 +3,7 @@ current_dir = meson.current_source_dir()
 cpu_family = host_machine.cpu_family()
 mod_features = import('features')
 
-HWY_SSE4_FLAGS = ['-DHWY_WANT_SSE4', '-DHWY_DISABLE_PCLMUL_AES']
+HWY_SSE4_FLAGS = ['-DHWY_WANT_SSE3', '-DHWY_DISABLE_PCLMUL_AES']
 # Use SSE for floating-point on x86-32 to ensure numeric consistency.
 # The x87 FPU's 80-bit internal precision causes unpredictable rounding
 # and overflow behavior when converting to smaller types. SSE maintains
@@ -11,10 +11,17 @@ HWY_SSE4_FLAGS = ['-DHWY_WANT_SSE4', '-DHWY_DISABLE_PCLMUL_AES']
 X86_64_V2_FLAGS = cpu_family == 'x86'? ['-mfpmath=sse'] : ['-mcx16']
 X86_64_V2_NAMES = cpu_family == 'x86'? [] : ['CX16']
 X86_V2 = mod_features.new(
-  'X86_V2', 1, args: ['-msse', '-msse2', '-msse3', '-mssse3', '-msse4.1', '-msse4.2',
-                      '-mpopcnt', '-msahf'] + X86_64_V2_FLAGS + HWY_SSE4_FLAGS,
+  'X86_V2', 1, args: ['-msse', '-msse2'
+#, '-msse3'
+#, '-mssse3'
+#, '-msse4.1', '-msse4.2',
+#                      '-mpopcnt', '-msahf'
+] + X86_64_V2_FLAGS + HWY_SSE4_FLAGS,
   # Adds compiler definitions `NPY_HAVE_SSE*` 
-  group: ['SSE', 'SSE2', 'SSE3', 'SSSE3', 'SSE41', 'SSE42', 'POPCNT', 'LAHF'] + X86_64_V2_NAMES,
+  group: ['SSE', 'SSE2'
+#, 'SSE3'
+#, 'SSSE3', 'SSE41', 'SSE42', 'POPCNT', 'LAHF'
+] + X86_64_V2_NAMES,
   detect: 'X86_V2',
   test_code: files(current_dir + '/test_x86_v2.c')[0],
 )
-- 
2.39.5

